Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/Controllers/MultiPositionController.cs

```diff
@@ -0,0 +1,1001 @@
+                    {
+                        Locomotive.ThrottlePercent = 100;
+                    }
+
+                }
+                if (controllerPosition == ControllerPosition.DynamicBrakeIncrease)
+                {
+                    if (CanControlTrainBrake)
+                    {
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "apply")
+                        {
+                            Locomotive.StartTrainBrakeDecrease(null);
+                        }
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "neutral")
+                        {
+                            Locomotive.StopTrainBrakeDecrease();
+                        }
+                    }
+                    if (Locomotive.DynamicBrakePercent == -1) Locomotive.SetDynamicBrakePercent(0);
+                    if (Locomotive.ThrottlePercent < 1 && Locomotive.DynamicBrakePercent < 100)
+                    {
+                        Locomotive.SetDynamicBrakePercent(Locomotive.DynamicBrakePercent + 2f);
+                    }
+                }
+                if (controllerPosition == ControllerPosition.DynamicBrakeIncreaseFast)
+                {
+                    if (CanControlTrainBrake)
+                    {
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "apply")
+                        {
+                            Locomotive.StartTrainBrakeDecrease(null);
+                        }
+                        if (Locomotive.TrainBrakeController.GetStatus().ToLower() == "neutral")
+                        {
+                            Locomotive.StopTrainBrakeDecrease();
+                        }
+                    }
+                    if (Locomotive.DynamicBrakePercent == -1) Locomotive.SetDynamicBrakePercent(0);
+                    if (Locomotive.ThrottlePercent < 1 && Locomotive.DynamicBrakePercent < 100)
+                    {
+                        Locomotive.SetDynamicBrakePercent(Locomotive.DynamicBrakePercent + 2f);
+                    }
+                }
+                if (controllerPosition == ControllerPosition.DynamicBrakeDecrease)
+                {
+                    if (Locomotive.DynamicBrakePercent > 0)
+                    {
+                        Locomotive.SetDynamicBrakePercent(Locomotive.DynamicBrakePercent - 1);
+                    }
+                }
+                if (controllerPosition == ControllerPosition.Drive || controllerPosition == ControllerPosition.ThrottleHold)
+                {
+                    if (Locomotive.DynamicBrakePercent < 2)
+                    {
+                        Locomotive.SetDynamicBrakePercent(-1);
+                    }
+                    if (Locomotive.DynamicBrakePercent > 1)
+                    {
+                        Locomotive.SetDynamicBrakePercent(Locomotive.DynamicBrakePercent - 1);
```

Comment:
Explain the reason of the 1 constant. Create a global variable, since this appears many times.

Reply:
done